soroban-sdk/src/crypto/bls12_381.rs

```diff
@@ -0,0 +1,294 @@
+use crate::{
+    env::internal::{self, BytesObject, U64Val},
+    impl_bytesn_repr,
+    unwrap::{UnwrapInfallible, UnwrapOptimized},
+    Bytes, BytesN, ConversionError, Env, IntoVal, TryFromVal, Val, Vec, U256,
+};
+use core::{cmp::Ordering, fmt::Debug};
+/// Bls12_381 provides access to curve and field arithmetics on the BLS12-381
+/// curve.
+pub struct Bls12_381 {
+    env: Env,
+}
+
+/// # `G1Affine` is a point in the G1 group (subgroup defined over the base field
```

Comment:
It's not common to make the first line of these doc comments a heading. Is there a reason we need it to be?
```suggestion
/// `G1Affine` is a point in the G1 group (subgroup defined over the base field
```